### ISSUE TITLE
`terraform test`: Modules with no resources aren't saving their output values

### ISSUE THREAD

**Issue by liamcervante** [hashicorp/terraform#34476]
### Terraform Version

```shell
1.7.0-rc1
1.6.6
```

### Terraform Configuration Files

```terraform
# main.tf

module "first" {
  source = "./first"
}
```

```terraform
# first/main.tf

module "second" {
    source = "../second"
}

output "id" {
    value = module.second.id
}
```

```terraform
# second/main.tf

resource "tfcoremock_simple_resource" "resource" {}

output "id" {
  value = tfcoremock_simple_resource.resource.id
}
```

```terraform
# main.tftest.hcl

run "run1" {
  command = apply

  assert {
    condition     = module.first.id != 0
    error_message = "Fail"
  }
}
```

### Debug Output

N/A

### Expected Behavior

test should pass

### Actual Behavior

test fails

### Steps to Reproduce

`terraform init`
`terraform test`

### References

- https://discuss.hashicorp.com/t/terraform-test-1-6-x-potential-bug-when-command-apply-unknown-condition-value/61363

**Comment by liamcervante** [comment]
Specifically, our definition of an empty module here: https://github.com/hashicorp/terraform/blob/main/internal/terraform/node_module_expand.go#L220 is causing the middle module in the configuration to be removed, and then the test assertion doesn't have access to the output from that module and so it returns `unknown`.

Interestingly, this is already fixed in `main` I think by this PR: https://github.com/hashicorp/terraform/pull/34343, which wasn't backported to `v1.7`.

**Comment by liamcervante** [comment]
This should be fixed in the next release of the v1.7 series.
